### PR TITLE
BE: RBAC: Impl separate permissions for topic analysis

### LINKED ISSUE #512: BE: RBAC: Impl separate permissions for topic analysis
### Issue submitter TODO list

- [X] I've searched for an already existing issues [here](https://github.com/kafbat/kafka-ui/issues)
- [X] I'm running a supported version of the application which is listed [here](https://github.com/kafbat/kafka-ui/blob/main/.github/SECURITY.md) and the feature is not present there

### Is your proposal related to a problem?

_No response_

### Describe the feature you're interested in

Currently, any user authorized to view the topic can run the analysis. I believe this feature should be more restricted.

### Describe alternatives you've considered

_No response_

### Version you're running

8c7012679d7ec0293e0f96a1740690bf0bfc994e

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bachmanity1**
Resolves #512. 

- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [ ] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [ ] My changes generate no new warnings (e.g. Sonar is happy)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

Check out [Contributing](https://github.com/kafbat/kafka-ui/blob/main/.github/CONTRIBUTING.md) and [Code of Conduct](https://github.com/kafbat/kafka-ui/blob/main/.github/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by bachmanity1** [comment]
Hi @Haarolean , can you have a look please?

**Review by Haarolean — CHANGES_REQUESTED** [review]
We'd need changes on the backend as well (see `TopicsController` class)

File: api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java [hunk 1]
```diff
@@ -13,6 +13,8 @@ public enum TopicAction implements PermissibleAction {
   MESSAGES_READ(VIEW),
   MESSAGES_PRODUCE(VIEW),
   MESSAGES_DELETE(VIEW, EDIT),
+  ANALYSIS_VIEW(VIEW),
```
Comment (Haarolean) [inline api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java:16]: This one is unused atm, we'd need to add this to both backend and frontend

**Review by Haarolean — CHANGES_REQUESTED** [review]

File: api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java (hunk 1 above)
Comment (bachmanity1) [reply api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java]: Couldn't find where to add it in the frontend.

Comment (Haarolean) [reply api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java]: Statistics tab should be disabled if there are no corresponding permissions

Comment (bachmanity1) [reply api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java]: @Haarolean PR seems ready, can you have a look?

File: api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java [hunk 2]
```diff
@@ -13,6 +13,8 @@ public enum TopicAction implements PermissibleAction {
   MESSAGES_READ(VIEW),
   MESSAGES_PRODUCE(VIEW),
   MESSAGES_DELETE(VIEW, EDIT),
+  ANALYSIS_VIEW(VIEW, MESSAGES_READ),
```
Comment (Haarolean) [inline api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java:16]: I believe that both permissions here shouldn't have messages-related permissions here.
Being able to run analysis doesn't imply that the person should be able to read messages.
Besides that, the rest looks good now.

**Review by Haarolean — CHANGES_REQUESTED** [review]

File: api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java (hunk 2 above)
Comment (bachmanity1) [reply api/src/main/java/io/kafbat/ui/model/rbac/permission/TopicAction.java]: done

**Comment by bachmanity1** [comment]
@Haarolean can we get this merged?

**Comment by Haarolean** [comment]
> @Haarolean can we get this merged?

as soon as we figure out the backward compatibility drawbacks

**Review by Haarolean — APPROVED** [review]

**Comment by Haarolean** [comment]
@bachmanity1 thanks for your first contribution to kafbat UI!

**Comment by yermulnik** [comment]
To double check I'm not missing something non-obvious: these two new permissions were not added to https://kafbat.io/configuration/rbac-role-based-access-control#permissions by an oversight or on purpose? Should I raise an issue to get docu updated? Thanks.

**Comment by bachmanity1** [comment]
> To double check I'm not missing something non-obvious: these two new permissions were not added to https://kafbat.io/configuration/rbac-role-based-access-control#permissions by an oversight or on purpose? Should I raise an issue to get docu updated? Thanks.

Yes, this was an oversight from my side. I think docs should be updated.

**Comment by Haarolean** [comment]
@yermulnik you can even submit a PR: https://github.com/kafbat/ui-docs :-)

**Comment by yermulnik** [comment]
> @yermulnik you can even submit a PR: [kafbat/ui-docs](https://github.com/kafbat/ui-docs?rgh-link-date=2025-02-05T08%3A36%3A12Z) :-)

NP 👍🏻 https://github.com/kafbat/ui-docs/pull/37

**Pull request merged**